src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildALibrary.cs

```diff
@@ -841,6 +841,62 @@ public void It_marks_package_references_as_externally_resolved(bool? markAsExter
+        [InlineData("net5.0", true, false, false, "True,False")]    // Pre .NET 6.0 predefinedCulturesOnly always false (no exist)
+        [InlineData("net5.0", false, true, true, "False,False")]    // Pre .NET 6.0 predefinedCulturesOnly always false (no exist)
+        [InlineData("net5.0", true, true, true, "True,False")]      // Pre .NET 6.0 predefinedCulturesOnly always false (no exist)
+        [InlineData("net6.0", false, false, false, "False,False")]  // predefinedCulturesOnly default value is false when Invariant is false or not defined
+        [InlineData("net6.0", false, false, true, "False,False")]   // predefinedCulturesOnly explicitly defined as false.
+        [InlineData("net6.0", false, true, true, "False,True")]     // predefinedCulturesOnly explicitly defined as true.
+        [InlineData("net6.0", true, false, false, "True,True")]     // predefinedCulturesOnly default value is true when Invariant is true
+        [InlineData("net6.0", true, false, true, "True,False")]     // predefinedCulturesOnly explicitly defined as false.
+        [InlineData("net6.0", true, true, true, "True,True")]       // predefinedCulturesOnly explicitly defined as true.
+        public void It_can_implicitly_define_predefined_Cultures_only(string targetFramework, bool invariantValue, bool predefinedCulturesOnlyValue, bool definePredefinedCulturesOnly, string expectedPredefinedCulturesOnlyValue)
+        {
+            var testProj = new TestProject()
+            {
+                Name = "CheckPredefineCulturesOnly",
+                TargetFrameworks = targetFramework,
+                IsExe = true,
+            };
+
+            testProj.AdditionalProperties["InvariantGlobalization"] = invariantValue ? "true" : "false";
+
+            if (definePredefinedCulturesOnly)
+            {
+                testProj.AdditionalProperties["PredefinedCulturesOnly"] = predefinedCulturesOnlyValue ? "true" : "false";
+            }
+
+            var testAsset = _testAssetsManager.CreateTestProject(testProj, targetFramework);
+            File.WriteAllText(Path.Combine(testAsset.Path, testProj.Name, $"{testProj.Name}.cs"), @"
+                using System;
+                using System.Reflection;
+                class Program
+                {
+                    static void Main(string[] args)
+                    {
+                        bool invariant = false;
+                        bool predefinedCulturesOnly = false;
+                        try { invariant = (bool) typeof(object).Assembly.GetType(""System.Globalization.GlobalizationMode"").GetProperty(""Invariant"", BindingFlags.Static | BindingFlags.NonPublic).GetValue(null); } catch {}
+                        try { predefinedCulturesOnly = (bool) typeof(object).Assembly.GetType(""System.Globalization.GlobalizationMode"").GetProperty(""PredefinedCulturesOnly"", BindingFlags.Static | BindingFlags.NonPublic).GetValue(null); } catch {}
```

Comment:
Is there a better way to check this than by using private reflection?  Would reading the runtimeconfig file work?